yt_dlp/extractor/txxx.py

```diff
@@ -0,0 +1,477 @@
+            'duration': parse_duration(traverse_obj(video_info, ('video', 'duration'))),
+            'view_count': int_or_none(traverse_obj(video_info, ('video', 'statistics', 'viewed'))),
+            'like_count': int_or_none(traverse_obj(video_info, ('video', 'statistics', 'likes'))),
+            'dislike_count': int_or_none(traverse_obj(video_info, ('video', 'statistics', 'dislikes'))),
+            'age_limit': 18,
+            'formats': formats,
+        }
+
+
+class PornTopIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?porntop\.com/video/(?P<id>\d+)(?:/(?P<display_id>[^/?]+))?'
+    _TESTS = [{
+        'url': 'https://porntop.com/video/101569/triple-threat-with-lia-lor-malena-morgan-and-dani-daniels/',
+        'md5': '612ba7b3cb99455b382972948e200b08',
+        'info_dict': {
+            'id': '101569',
+            'display_id': 'triple-threat-with-lia-lor-malena-morgan-and-dani-daniels',
+            'ext': 'mp4',
+            'title': 'Triple Threat With Lia Lor, Malena Morgan And Dani Daniels',
+            'description': 'md5:c511b7de2d6135c14868f0e6b0940c41',
+            'uploader': 'PatrickBush',
+            'duration': 480,
+            'view_count': int,
+            'like_count': int,
+            'dislike_count': int,
+            'age_limit': 18,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id, display_id = self._match_valid_url(url).group('id', 'display_id')
+
+        webpage = self._download_webpage(url, video_id)
+
+        # find the VideoObject json object
+        video_obj_text = self._search_regex(
+            r'<script[^>]*>[^<]*schemaJson\s*=\s*(?P<json_ld>[^<]+VideoObject[^<]+)\s*;\s*var\s+script\s*=[^<]*</script>',
+            webpage, 'VideoObject', group='json_ld')
+        # there are javascript code within the declaration that would break json parsing
+        # the statistics values look like this: parseInt("2697"). remove the parseInt() function
+        video_obj_text = re.sub(r'parseInt\([^\d]+(\d+)[^\d]+\)', r'\1', video_obj_text)
+        # remove the function at "duration"
+        video_obj_text = re.sub(r'\(function\(duration\).*\(([^)]+)\)(\s*,\s*"thumbnailUrl")', r'\1\2', video_obj_text)
+        # change single quote to double quote
+        video_obj_text = video_obj_text.replace("'", '"')
+        # parse the string
+        video_obj = self._parse_json(video_obj_text, video_id, fatal=True)
+
+        views = 0
+        likes = 0
+        dislikes = 0
+        stats_obj = traverse_obj(video_obj, 'interactionStatistic', ...)
+        for stat in stats_obj:
+            if stat['interactionType'] == 'http://schema.org/WatchAction':
+                views = stat['userInteractionCount']
+            elif stat['interactionType'] == 'http://schema.org/LikeAction':
+                likes = stat['userInteractionCount']
+            elif stat['interactionType'] == 'http://schema.org/DislikeAction':
+                dislikes = stat['userInteractionCount']
```

Comment:
self._search_json_ld doesn't quite work here because their structure does not follow the pattern in JSON_LD_RE. Also, there is JavaScript code within the data.

Reply:
Updated code to use self._json_ld to load values after capturing and fixing the sub-string.